test/integration/http_integration.cc

```diff
@@ -2117,6 +2117,287 @@ void HttpIntegrationTest::testTrailers(uint64_t request_size, uint64_t response_
   }
 }
 
+// Adds metadata inserting filter before metadata consuming filter. Verify metadata are consumed.
+void HttpIntegrationTest::testInsertBeforeConsumeResponseMetadata() {
+  config_helper_.addFilter(R"EOF(
```

Comment:
I'd lean towards either pre-declaring these strings or having utility functions for addMetadataInsertFilter etc.

Reply:
Good idea! Done.